packages/fuels-core/src/types/wrappers/transaction_response.rs

```diff
@@ -27,8 +27,11 @@ pub enum TransactionStatus {
     SqueezedOut(),
 }
 
-impl From<ClientTransactionResponse> for TransactionResponse {
-    fn from(client_response: ClientTransactionResponse) -> Self {
+impl TransactionResponse {
+    pub fn from_fuel_response(
+        client_response: ClientTransactionResponse,
+        consensus_parameters: ConsensusParameters,
```

Comment:
Could this lead to bugs? I imagine a scenario where the params changed at some point T during the growth of the chain. Now, some time after T a user wants to inspect a tx that happened before T. They retrieve the tx but the attached params correspond to the current ones instead of the ones at the time of execution. 

I don't suppose there is a realistic scenario where `chain_id` could change so calling `id` should be safe regardless.  However, calling `fee_checked_from_tx` could lead to wrong results. It's a weird use case to calculate the fee on an old tx but still a bug.

I'm not yet sure how we can tackle this but for now I just wanted to confirm the above assumptions are true.

Reply:
Good point. I do not see a better way to deal with this for now so I have reverted both `fee_checked_from_tx` and `id` to accept `ConsensusParameters` and `ChainId` respectively.